lib/cosmos/packet_logs/packet_log_writer.rb

```diff
@@ -198,17 +226,29 @@ def start_new_file
       end
       @start_time = Time.now
       Logger.instance.info "Log File Opened : #{@filename}"
-      start_new_file_hook()
+      start_new_file_hook(packet)
     rescue => err
       Logger.instance.error "Error opening #{@filename} : #{err.formatted}"
       @logging_enabled = false
       Cosmos.handle_critical_exception(err)
     end
 
-    # Hook after writing the file header in start_new_file
+    # Adds the meta packet at the beginning of telemetry packet logs
     # Mutex is held during this hook
-    def start_new_file_hook
-      # Default do nothing
+    def start_new_file_hook(packet)
+      # If the first packet is a SYSTEM META packet, make sure the file header matches
```

Comment:
write_packet(packet), can call start_new_file(packet), which calls start_new_file_hook(packet).  If that packet is already a SYSTEM META packet, I want to update the file header and guarantee the correct configuration name is put there.   This is more of an issue for the database logging than than CmdTlmServer, but it could happen.